### PR TITLE
Change UI to be more similar to current extension: Transactions List

### PR DESCRIPTION
**Pull request by buberdds**
mobile
Screenshot 2022-07-21 at 15 10 47
Screenshot 2022-07-21 at 15 11 05

desktop
Screenshot 2022-07-21 at 15 10 34
Screenshot 2022-07-21 at 15 10 25

### CODE REVIEW

File: src/app/components/Transaction/index.tsx [hunk 1]
```diff
@@ -35,7 +37,7 @@ export enum TransactionSide {
 type TransactionDictionary = {
   [type in transactionTypes.TransactionType]: {
     [side in TransactionSide]: {
-      icon: () => React.ReactNode
+      icon: React.ElementType
```
Comment (lukaw3d) [inline src/app/components/Transaction/index.tsx:40]: Use grommet's `icon: Icon` type so `<Icon size={isMobile ? '20px' : 'medium'} color="brand" />` type-check props

File: src/app/components/Transaction/index.tsx [hunk 2]
```diff
@@ -221,67 +224,115 @@ export function Transaction(props: TransactionProps) {
...
           )}
-          {!otherAddress && (
-            <InfoBox
-              icon={<ContactInfo color="brand" />}
-              label={designation}
-              value={t('common.unavailable', 'Unavailable')}
-            />
+
+          {!isMobile && (
+            <Grid columns={{ count: 'fit', size: 'xsmall' }} gap="none">
```
[the remark this answers is not in the document]
Comment (buberdds) [reply src/app/components/Transaction/index.tsx:288]: Screenshot 2022-07-23 at 17 50 25

I've added a custom breakpoint in this view and a comment about what we need to do if we want to move this into theme config. After the update amount section will push other content down between 768px-1024px.

**Review by lukaw3d — APPROVED** [review]

File: src/app/components/Transaction/index.tsx [hunk 3]
```diff
@@ -221,67 +224,115 @@ export function Transaction(props: TransactionProps) {
...
+              <Box pad="none">
+                {otherAddress && (
+                  <NavLink data-testid="external-wallet-address" to={`/account/${otherAddress}`}>
```
[the remark this answers is not in the document]
Comment (tjanez) [reply src/app/components/Transaction/index.tsx:291]: Agreed, I think it should not open it but rather copy the address to the clipboard.

Same for Tx hash, I would copy it to clipboard upon pressing on it.

File: src/app/components/Transaction/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tjanez) [reply src/app/components/Transaction/index.tsx:288]: Looks better now, thanks!

File: src/app/components/Transaction/index.tsx [hunk 4]
```diff
@@ -221,67 +224,115 @@ export function Transaction(props: TransactionProps) {
     ? transactionDictionary[transaction.type][side]
     : unrecognizedTransaction
 
-  const icon = matchingConfiguration.icon()
+  const Icon = matchingConfiguration.icon
   const header = matchingConfiguration.header()
   const designation = matchingConfiguration.designation
```
[the remark this answers is not in the document]
Comment (tjanez) [reply src/app/components/Transaction/index.tsx:246]: @lukaw3d, do we need to use `icon` instead of `Icon`?

**Review by tjanez — APPROVED** [review]
Product-wise looks good!

File: src/app/components/Transaction/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/components/Transaction/index.tsx:246]: No, I mean "designation"

File: src/app/components/Transaction/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (buberdds) [reply src/app/components/Transaction/index.tsx]: Added copy to clipboard feature. I will make PR to update Grommet Tip in my free time so user will get a proper feedback after clicking addr or tx hash.

[the remark this answers is not in the document]
Comment (tjanez) [reply src/app/components/Transaction/index.tsx]: Ok, great :slightly_smiling_face:

File: src/app/components/Transaction/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tjanez) [reply src/app/components/Transaction/index.tsx:245]: Ok, I still don't get it :slightly_smiling_face:.

@lukaw3d, can you file a follow-up issue to rename "designation" to your liking :slightly_smiling_face:.

[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/components/Transaction/index.tsx:245]: yep yep https://github.com/oasisprotocol/oasis-wallet-web/issues/943

File: src/app/components/ShortAddress/__tests__/index.test.tsx [hunk 5]
```diff
@@ -1,13 +1,35 @@
...
+  })
+
+  it('should return short hash', () => {
+    expect(trimLongString('6557c86267761b845aa00dac39e892ca58922305c883965f4b0b67a009b0d8d3')).toEqual(
+      '6557c86267...09b0d8d3',
+    )
+  })
+
+  it('should return original string if it is shorter than trimStart value', () => {
+    expect(trimLongString('foo')).toEqual('foo')
```
Comment (lukaw3d) [inline src/app/components/ShortAddress/__tests__/index.test.tsx:18]: This isn't generalized for short strings yet:
`trimLongString('Unavailable')` returns `'Unavailabl...vailable'`

File: src/app/components/Transaction/InfoBox.tsx [hunk 6]
```diff
@@ -1,23 +1,34 @@
...
-          {props.icon}
-        </Box>
-      )}
+    <Box
+      direction="row"
+      gap="small"
+      hoverIndicator={{ color: 'background-contrast' }}
+      pad={{ horizontal: 'small', vertical: 'small' }}
+      onClick={copyToClipboard ? () => copy(value) : undefined}
+      style={{ boxShadow: 'none' }}
```
Comment (lukaw3d) [inline src/app/components/Transaction/InfoBox.tsx:23]: This is terrible for keyboard accessibility

File: src/app/components/Toolbar/Features/AccountSelector/__tests__/__snapshots__/index.test.tsx.snap [hunk 7]
```diff
@@ -479,7 +479,7 @@ exports[`<AccountSelector  /> should match snapshot 1`] = `
                       <span
                         class="c11"
                       >
-                        dummy...dummy
```
Comment (lukaw3d) [inline src/app/components/Toolbar/Features/AccountSelector/__tests__/__snapshots__/index.test.tsx.snap:482]: :facepalm:

File: src/app/components/Transaction/InfoBox.tsx (hunk 6 above)
Comment (buberdds) [reply src/app/components/Transaction/InfoBox.tsx:23]: hm indeed that sucks. will fix that tomorrow.

File: src/app/components/Transaction/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tjanez) [reply src/app/components/Transaction/index.tsx:245]: Ok, now I get it :slightly_smiling_face:. Thanks, @lukaw3d for filling the issue and @buberdds for fixing it already :slightly_smiling_face: !

File: src/app/components/ShortAddress/__tests__/index.test.tsx (hunk 5 above)
Comment (lukaw3d) [reply src/app/components/ShortAddress/__tests__/index.test.tsx:18]: https://github.com/oasisprotocol/oasis-wallet-web/issues/950

**Pull request merged**